### DISCUSSION TITLE
MDX: Incompatible remark plugins

### DISCUSSION THREAD

**Discussion by laurieontech** [gatsbyjs/gatsby#25066] [Umbrella Discussions]
Presently, `gatsby-plugin-mdx` takes an optional array of `remarkPlugins`. While this is meant to support all remark compatible plugins there are situations where it does not and it breaks MDX support in a site entirely.

Solving this is likely a multipronged approach. There are specific, popular plugins that need to always work and require test coverage to validate that moving forward. For other plugins it may make sense to ignore them with a warning and continue to allow the site to build.

Currently exploring potential resolutions.

## Related bugs/issues

- [vscode and prism remark plugin performance](https://github.com/gatsbyjs/gatsby/issues/17443)
- https://github.com/gatsbyjs/gatsby/issues/16983
- https://github.com/gatsbyjs/gatsby/issues/25013

**Comment by mariusGundersen** [comment]
I've stumbled onto this using just mdx and rehype-mathjax. The problem shows up as all the generated html having `classname=""` attributes rather than `class=""` attributes.

**Comment by phun-ky** [comment]
I do not know if this is the correct place for this, but using `gatsby-remark-images` with markdown files using `img`-tags, breaks layout (closes markup unexpectedly early). Experienced this first a long time ago, and let it slide, since it did not break my pages, it just looked a bit ugly. However, using `gatsy-remark-images` with `gatsby-plugin-mdx` breaks the dev build in console, due to unexpecting closing tag.
